src/controllers/legacycontrollersettings.h

```diff
@@ -165,8 +165,8 @@ class LegacyControllerBooleanSetting
         emit valueReset();
     }
 
-    static AbstractLegacyControllerSetting* createFrom(const QDomElement& element) {
-        return new LegacyControllerBooleanSetting(element);
+    static std::shared_ptr<LegacyControllerBooleanSetting> createFrom(const QDomElement& element) {
+        return std::make_shared<LegacyControllerBooleanSetting>(element);
```

Comment:
slight nitpick: creating a `shared_ptr` from a `unique_ptr` is easy, the other way around is hard. So unless we are sure that all consumers will always need shared ownership, prefer that first... I'll leave it up to you to make that evaluation.

Reply:
In Mixxx we always need a shared_ptr. We don't have shared pointers in tests. 
I like to benefit here form `std::make_shared` which allocates the object and the control structure for the shared pointer in one go.

Reply:
I know, questionable whether thats really needed and whether that assumption will really be true in the future too, but I won't insist.

Reply:
I did a closer look. We require a std::share pointer because the AbstractLegacyControllerSetting is part of the mapping which is copied to the preferences (main thread) altered and and than moved back to the Controller thread. 

Copying all the control structures along with it feels wrong, however changing this is another refactoring, at least out of scope of this PR.